File: Blanquita-Backend/app/Models/BobinaPapel/InventarioBobinaPapel.py
from datetime import date, datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class VerResumenInventarioBobinaPapelResponse(BaseModel):
    IdTipoBobina: int
    NombreTipoBobina: str
    CantidadBobinas: int
    PesoNetoTotalKg: Decimal
    GramajePromedio: Decimal


class VerDetalleInventarioBobinaPapelRequest(BaseModel):
    IdTipoBobina: int


class VerDetalleInventarioBobinaPapelResponse(BaseModel):
    IdBobinaPapel: int
    CodigoBobina: str
    CodigoLote: str
    FechaRecepcion: date
    NombreProveedor: str
    PesoBrutoKg: Decimal
    PesoNetoKg: Decimal
    Gramaje: Decimal


class ReingresarBobinaAInventarioRequest(BaseModel):
    IdBobinaPapel: int
    Observacion: Optional[str] = None


class ReingresarBobinaAInventarioResponse(BaseModel):
    IdBobinaPapel: int
    IdEstadoMateriaPrima: int
    FechaMovimiento: datetime


class DarDeBajaBobinaRequest(BaseModel):
    IdBobinaPapel: int
    Observacion: Optional[str] = None


class DarDeBajaBobinaResponse(BaseModel):
    IdBobinaPapel: int
    IdEstadoMateriaPrima: int
    FechaMovimiento: datetime


class VerBobinasPapelFueraInventarioResponse(BaseModel):
    IdBobinaPapel: int
    CodigoBobina: str
    NombreTipoBobina: str
    PesoBrutoKg: Optional[float]
    Gramaje: Optional[float]
    NombreProveedor: str
    FechaRecepcion: date
    UltimaObservacion: Optional[str]
    FechaUltimoMovimiento: Optional[datetime]

File: Blanquita-Frontend/src/models/BobinaPapel/InventarioBobinaPapel.js
export class VerResumenInventarioBobinaPapelResponse {
  constructor({ IdTipoBobina, NombreTipoBobina, CantidadBobinas, PesoNetoTotalKg, GramajePromedio }) {
    this.IdTipoBobina = IdTipoBobina;
    this.NombreTipoBobina = NombreTipoBobina;
    this.CantidadBobinas = CantidadBobinas;
    this.PesoNetoTotalKg = PesoNetoTotalKg;
    this.GramajePromedio = GramajePromedio;
  }
}

export class VerDetalleInventarioBobinaPapelRequest {
  constructor({ IdTipoBobina }) {
    this.IdTipoBobina = IdTipoBobina;
  }
}

export class VerDetalleInventarioBobinaPapelResponse {
  constructor({
    IdBobinaPapel,
    CodigoBobina,
    CodigoLote,
    FechaRecepcion,
    NombreProveedor,
    PesoBrutoKg,
    PesoNetoKg,
    Gramaje,
  }) {
    this.IdBobinaPapel = IdBobinaPapel;
    this.CodigoBobina = CodigoBobina;
    this.CodigoLote = CodigoLote;
    this.FechaRecepcion = FechaRecepcion;
    this.NombreProveedor = NombreProveedor;
    this.PesoBrutoKg = PesoBrutoKg;
    this.PesoNetoKg = PesoNetoKg;
    this.Gramaje = Gramaje;
  }
}